test/nodetool/test_compact.py

```diff
@@ -24,17 +22,14 @@ def test_keyspace(nodetool):
 
 
 def test_nonexistent_keyspace(nodetool):
-    with pytest.raises(subprocess.CalledProcessError) as e:
-        nodetool("compact", "non_existent_ks", expected_requests=[
+    utils.check_nodetool_fails_with(
+            nodetool,
+            ("compact", "non_existent_ks"),
+            {"expected_requests": [
                 expected_request("GET", "/storage_service/keyspaces", multiple=True, response=["system"]),
-                expected_request("POST", "/storage_service/keyspace_compaction/non_existent_ks")])
-
-    err_lines = e.value.stderr.rstrip().split('\n')
-    out_lines = e.value.stdout.rstrip().split('\n')
-    cassandra_error = "nodetool: Keyspace [non_existent_ks] does not exist."
-    scylla_error = "error processing arguments: keyspace non_existent_ks does not exist"
-    assert cassandra_error in err_lines or cassandra_error in out_lines or \
-        scylla_error in err_lines or scylla_error in out_lines
+                expected_request("POST", "/storage_service/keyspace_compaction/non_existent_ks")]},
+            "nodetool: Keyspace [non_existent_ks] does not exist.",
```

Comment:
Style nitpick:  it will be hard to remember what the two different messages mean. In python you can put option names when it's ambiguous, e.g., `check_nodetool_fails_with(..., scylla_error='...', cassandra_error='...')` and it's clearer to read.

Reply:
I will change it to be a list instead.

Reply:
Done in v4.